internal/services/resource/resource_management_private_link_association_resource_test.go

```diff
@@ -0,0 +1,152 @@
+	data := acceptance.BuildTestData(t, "azurerm_resource_management_private_link_association", "test")
+	r := ResourceManagementPrivateLinkAssociationTestResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r ResourceManagementPrivateLinkAssociationTestResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := privatelinkassociation.ParsePrivateLinkAssociationID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Resource.PrivateLinkAssociationClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("reading %s: %+v", *id, err)
+	}
+
+	return utils.Bool(resp.Model != nil), nil
+}
+
+func (r ResourceManagementPrivateLinkAssociationTestResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_resource_management_private_link_association" "test" {
+  management_group_id                 = data.azurerm_management_group.test.id
+  resource_management_private_link_id = azurerm_resource_management_private_link.test.id
+  public_network_access_enabled       = true
+}
+`, r.template(data))
+}
+
+func (r ResourceManagementPrivateLinkAssociationTestResource) name(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+%s
+
+resource "azurerm_resource_management_private_link_association" "test" {
+  name                                = "12345678-1234-1234-1234-123456789012"
+  management_group_id                 = data.azurerm_management_group.test.id
+  resource_management_private_link_id = azurerm_resource_management_private_link.test.id
+  public_network_access_enabled       = true
+}
+`, r.template(data))
+}
+
+func (r ResourceManagementPrivateLinkAssociationTestResource) requiresImport(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_resource_management_private_link_association" "import" {
+  name                                = azurerm_resource_management_private_link_association.test.name
+  management_group_id                 = azurerm_resource_management_private_link_association.test.management_group_id
+  resource_management_private_link_id = azurerm_resource_management_private_link_association.test.resource_management_private_link_id
+  public_network_access_enabled       = azurerm_resource_management_private_link_association.test.public_network_access_enabled
+}
+`, r.basic(data))
+}
+
+func (r ResourceManagementPrivateLinkAssociationTestResource) template(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+variable "primary_location" {
+  default = %q
+}
+variable "random_string" {
+  default = %q
+}
+variable "random_integer" {
+  default = %d
+}
```

Comment:
Given that these seem to be only used once, can we remove these and use the `data.Locations.Primary`, `data.RandomString` and `data.RandomInteger` directly where needed?

Reply:
fixed, thanks.